src/libraries/System.Runtime.Extensions/tests/System/OperatingSystemTests.cs

```diff
@@ -205,18 +205,11 @@ private static void TestIsOSPlatform(string currentOSName, Func<bool> currentOSC
             // MacCatalyst is a special case since it also returns true for iOS
             if (currentOSName == "MacCatalyst")
             {
-                Assert.Equal(10, allResults.Length);
-                Assert.False(allResults[0]); // IsBrowser()
-                Assert.False(allResults[1]); // IsLinux()
-                Assert.False(allResults[2]); // IsFreeBSD()
-                Assert.False(allResults[3]); // IsAndroid()
-                Assert.True(allResults[4]);  // IsIOS()
-                Assert.True(allResults[5]);  // IsMacCatalyst()
-                Assert.False(allResults[6]); // IsMacOS()
-                Assert.False(allResults[7]); // IsTvOS()
-                Assert.False(allResults[8]); // IsWatchOS()
-                Assert.False(allResults[9]); // IsWindows()
-                Assert.False(allResults[10]); // IsWasi()
+                for (int i = 0; i < allResults.Length; i++)
+                {
+                    bool expected = (i == 4 || i == 5) ? true : false; // true for IsIOS() and IsMacCatalyst()
```

Comment:
```suggestion
                    bool expected = i is 4 or 5; // true for IsIOS() and IsMacCatalyst()
```
?

Might be cleaner to change allResults to instead be a `Dictionary<string,bool>`, where the string is the name of the Is method.  The else case could still assert that only one value was true, and this case could assert that both "IsIOS" and "IsMacCatalyst" are true and everything else is false.

Reply:
Updated with your suggestion